File: 3740-minimum-distance-between-three-equal-elements-i/3740-minimum-distance-between-three-equal-elements-i.cpp
class Solution {
public:
    int minimumDistance(vector<int>& nums) {
        map<int,vector<int>>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]].push_back(i);
        }
        int mini = INT_MAX;
        for(auto i:mp){
            auto v = i.second;
            if(v.size()>=3){
                for(int j=2;j<v.size();j++){
                    mini = min(mini,abs(v[j]-v[j-1])+abs(v[j-1]-v[j-2])+abs(v[j]-v[j-2]));
                }
            }
        }
        return mini==INT_MAX?-1:mini;
    }
};

File: 3740-minimum-distance-between-three-equal-elements-i/3740-minimum-distance-between-three-equal-elements-i.java
import java.util.*;
class Solution {
    public int minimumDistance(int[] nums) {
        int n = nums.length;
        Map<Integer,List<Integer>> mp = new HashMap<>();
        for(int i = 0;i < n;i++){
            if(!mp.containsKey(nums[i])){
                mp.put(nums[i],new ArrayList<>());
            }
            mp.get(nums[i]).add(i);
        }
        Integer mini = Integer.MAX_VALUE;
        for(List<Integer> list:mp.values()){
            for(int j=2;j<list.size();j++){
                int a = list.get(j);
                int b = list.get(j-1);
                int c = list.get(j-2);
                int dist = Math.abs(a-b)+Math.abs(b-c)+Math.abs(a-c);
                mini = Math.min(mini,dist);
            }
        }
        return mini == Integer.MAX_VALUE?-1:mini;
    }
}